src/main/kotlin/org/rust/cargo/project/configurable/CargoConfigurable.kt

```diff
@@ -6,95 +6,82 @@
-
-    private val runExternalLinterOnTheFlyCheckbox: JBCheckBox = JBCheckBox()
-    private var runExternalLinterOnTheFly: Boolean by CheckboxDelegate(runExternalLinterOnTheFlyCheckbox)
-
-    private val useOfflineCheckbox: JBCheckBox = JBCheckBox()
-    private var useOffline: Boolean by CheckboxDelegate(useOfflineCheckbox)
-
-    private val compileAllTargetsCheckBox = JBCheckBox()
-    private var compileAllTargets: Boolean by CheckboxDelegate(compileAllTargetsCheckBox)
+class CargoConfigurable(project: Project) : RsConfigurableBase(project, "Cargo") {
 
     private lateinit var externalLinterArguments: RsCommandLineEditor
 
-    override fun getDisplayName(): String = "Cargo"
-
-    override fun createComponent(): JComponent = panel {
+    override fun createPanel(): DialogPanel {
         externalLinterArguments = RsCommandLineEditor(
-            project, CargoCommandCompletionProvider(project.cargoProjects, "check ") { null }
+            project,
+            CargoCommandCompletionProvider(project.cargoProjects, "check ") { null }
         )
-
-        row("Watch Cargo.toml:", """
-            Update project automatically if `Cargo.toml` changes.
-        """) { autoUpdateEnabledCheckbox() }
-        row("Compile all project targets if possible:", """
-            Pass `--target-all` option to cargo build/check command.
-        """) { compileAllTargetsCheckBox() }
-        row("Offline mode:", """
-            Pass `--offline` option to cargo not to perform network requests.
-        """) { useOfflineCheckbox() }
-
-        titledRow("External Linter") {
-            subRowIndent = 0
-            row("External tool:", """
-                External tool to use for code analysis.
-            """) { externalLinterComboBox(growX, pushX) }
-            row("Additional arguments:", """
-                Additional arguments to pass to `cargo check` or `cargo clippy` command.
-            """) { externalLinterArguments(growX, pushX) }
-            row("Run external linter to analyze code on the fly:", """
-                Enable external linter to add code highlighting based on the used linter result.
-                Can be CPU-consuming.
-            """) { runExternalLinterOnTheFlyCheckbox() }
+        return panel {
+            row {
+                checkBox(
+                    "Update project automatically if Cargo.toml changes",
+                    state::autoUpdateEnabled
+                )
+            }
+            row {
+                checkBox(
+                    "Compile all project targets if possible",
+                    state::compileAllTargets,
+                    comment = "Pass <b>--target-all</b> option to Сargo <b>build</b>/<b>check</b> command"
+                )
+            }
+            row {
+                checkBox(
+                    "Offline mode",
+                    state::useOffline,
+                    comment = "Pass <b>--offline</b> option to Cargo not to perform network requests"
```

Comment:
As I understand reading [Cargo --offline](https://www.ncameron.org/blog/cargo-offline/) blog post, the `--offline` option only disables crates index update. Maybe clarify it here?

Like `Pass --offline option to force Cargo not to update the crates registry`

Reply:
I'm not sure that `crates registry` is more understandable for users (especially for newbies in the ecosystem) than `not to perform network requests`.
For example, https://doc.rust-lang.org/cargo/faq.html#how-can-cargo-work-offline says nothing about crates registry as well